services/provider/client/client.go

```diff
@@ -10,13 +10,34 @@ import (
 )
 
 type OCSProviderClient struct {
-	Client  pb.OCSProviderClient
-	timeout time.Duration
+	Client     pb.OCSProviderClient
+	clientConn *grpc.ClientConn
+	timeout    time.Duration
 }
 
-// NewProviderClient creates a client to talk to OCS provider server
-func NewProviderClient(cc *grpc.ClientConn, timeout time.Duration) *OCSProviderClient {
-	return &OCSProviderClient{Client: pb.NewOCSProviderClient(cc), timeout: timeout}
+// NewProviderClient creates a client to talk to the external OCS storage provider server
+func NewProviderClient(serverAddr string, timeout time.Duration) (*OCSProviderClient, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
```

Comment:
Please pass the context from the outside. 
For the interim period (until we refactor `context` usage on the for reconcile loop)  we can just create a new context on the call site